pymc/logprob/abstract.py

```diff
@@ -169,3 +162,49 @@ def __init__(self, scalar_op, *args, **kwargs):
                 f"Acceptable types are {self.valid_scalar_types}"
             )
         super().__init__(scalar_op, *args, **kwargs)
+
+    def __str__(self):
+        return f"Measurable{super().__str__()}"
+
+
+class ValuedRV(Op):
+    r"""Represents the association of a measurable variable and its value.
+
+    A `ValuedVariable` node represents the pair :math:`(Y, y)`, where
+    :math:`Y` is a random variable and :math:`y \sim Y`.
+
+    Log-probability (densities) are functions over these pairs, which makes
```

Comment:
```suggestion
    The log-probability function takes such pairs as input, which makes
```
IIRC densities are only for continuous random variables. This is kind of nitpicky but you are the judge :)